### ISSUE TITLE
Missing commit information in just-built 3.0 SDK

### ISSUE THREAD

**Issue by omajid** [dotnet/source-build#1140]
I just built commit `d789550` in source-build. The generated SDK (`dotnet-sdk-3.0.100-preview6-012264-fedora.29-x64.tar.gz`) is missing commit information for corefx and coreclr:

```
$ find -iname '*.so' -exec sh -c "echo {} ; strings {} | grep '@(#)'" \;
./shared/Microsoft.NETCore.App/3.0.0-preview6-27804-01/System.Native.so
@(#)Version 4.700.19.26409 @Commit: unknown
./shared/Microsoft.NETCore.App/3.0.0-preview6-27804-01/libsos.so
@(#)No version information produced
./shared/Microsoft.NETCore.App/3.0.0-preview6-27804-01/libdbgshim.so
@(#)No version information produced
./shared/Microsoft.NETCore.App/3.0.0-preview6-27804-01/libcoreclr.so
@(#)No version information produced
./shared/Microsoft.NETCore.App/3.0.0-preview6-27804-01/System.Security.Cryptography.Native.OpenSsl.so
@(#)Version 4.700.19.26409 @Commit: unknown
./shared/Microsoft.NETCore.App/3.0.0-preview6-27804-01/libsosplugin.so
@(#)No version information produced
./shared/Microsoft.NETCore.App/3.0.0-preview6-27804-01/libcoreclrtraceptprovider.so
./shared/Microsoft.NETCore.App/3.0.0-preview6-27804-01/System.IO.Compression.Native.so
@(#)Version 4.700.19.26409 @Commit: unknown
./shared/Microsoft.NETCore.App/3.0.0-preview6-27804-01/libmscordaccore.so
@(#)No version information produced
@(#)No version information produced
@(#)No version information produced
@(#)No version information produced
@(#)No version information produced
./shared/Microsoft.NETCore.App/3.0.0-preview6-27804-01/System.Globalization.Native.so
@(#)No version information produced
./shared/Microsoft.NETCore.App/3.0.0-preview6-27804-01/libhostpolicy.so
@(#)Version 3.0.27804.01 @BuiltBy: omajid-trull @Branch: release/3.0 @SrcCode: https://github.com/dotnet/core-setup/tree/fdf81c6faf7c7e0463d191a3a1d36c25c201e5cb
./shared/Microsoft.NETCore.App/3.0.0-preview6-27804-01/libclrjit.so
@(#)No version information produced
./shared/Microsoft.NETCore.App/3.0.0-preview6-27804-01/System.Net.Http.Native.so
@(#)Version 4.700.19.26409 @Commit: unknown
./shared/Microsoft.NETCore.App/3.0.0-preview6-27804-01/System.Net.Security.Native.so
@(#)Version 4.700.19.26409 @Commit: unknown
./shared/Microsoft.NETCore.App/3.0.0-preview6-27804-01/libmscordbi.so
@(#)No version information produced
./packs/Microsoft.NETCore.App.Host.fedora.29-x64/3.0.0-preview6-27804-01/runtimes/fedora.29-x64/native/libnethost.so
@(#)Version 3.0.27804.01 @BuiltBy: omajid-trull @Branch: release/3.0 @SrcCode: https://github.com/dotnet/core-setup/tree/fdf81c6faf7c7e0463d191a3a1d36c25c201e5cb
./host/fxr/3.0.0-preview6-27804-01/libhostfxr.so
@(#)Version 3.0.27804.01 @BuiltBy: omajid-trull @Branch: release/3.0 @SrcCode: https://github.com/dotnet/core-setup/tree/fdf81c6faf7c7e0463d191a3a1d36c25c201e5cb
```

**Comment by omajid** [comment]
This looks like a repeat of #651

**Comment by crummel** [comment]
Once https://github.com/dotnet/source-build/pull/1124 is merged we'll be able to change around our method of embedding the build info to match how all our contributing repos do it, so I think that might take care of the issue (hopefully for good this time).

**Comment by omajid** [comment]
Possibly related: https://github.com/dotnet/core-setup/issues/7635

**Comment by dagood** [comment]
(Saw the backlink) If there's a common thread, maybe it's that sourcelink was being disabled--when Arcade generates the header file that sticks the versions into the binaries, it depends on sourcelink to figure out what the repo state is.

**Comment by dseefeld** [comment]
Is this fixed in Preview8?  source-link was added there.

**Comment by omajid** [comment]
It still seems to be missing for some files on my end:

```
/usr/lib64/dotnet/packs/Microsoft.NETCore.App.Host.linux-x64/3.0.0-preview8-28405-07/runtimes/linux-x64/native/libnethost.so
@(#)No version information produced
/usr/lib64/dotnet/host/fxr/3.0.0-preview8-28405-07/libhostfxr.so
@(#)No version information produced
/usr/lib64/dotnet/shared/Microsoft.NETCore.App/3.0.0-preview8-28405-07/System.Native.so
@(#)Version 4.700.19.40503 @Commit: 1bb52e6a3db7f3673a3825f3677b9f27b9af99aa
/usr/lib64/dotnet/shared/Microsoft.NETCore.App/3.0.0-preview8-28405-07/libcoreclr.so
@(#)No version information produced
/usr/lib64/dotnet/shared/Microsoft.NETCore.App/3.0.0-preview8-28405-07/System.IO.Compression.Native.so
@(#)Version 4.700.19.40503 @Commit: 1bb52e6a3db7f3673a3825f3677b9f27b9af99aa
/usr/lib64/dotnet/shared/Microsoft.NETCore.App/3.0.0-preview8-28405-07/libhostpolicy.so
@(#)No version information produced
/usr/lib64/dotnet/shared/Microsoft.NETCore.App/3.0.0-preview8-28405-07/System.Globalization.Native.so
@(#)No version information produced
/usr/lib64/dotnet/shared/Microsoft.NETCore.App/3.0.0-preview8-28405-07/System.Net.Http.Native.so
@(#)Version 4.700.19.40503 @Commit: 1bb52e6a3db7f3673a3825f3677b9f27b9af99aa
/usr/lib64/dotnet/shared/Microsoft.NETCore.App/3.0.0-preview8-28405-07/System.Net.Security.Native.so
@(#)Version 4.700.19.40503 @Commit: 1bb52e6a3db7f3673a3825f3677b9f27b9af99aa
/usr/lib64/dotnet/shared/Microsoft.NETCore.App/3.0.0-preview8-28405-07/libclrjit.so
@(#)No version information produced
/usr/lib64/dotnet/shared/Microsoft.NETCore.App/3.0.0-preview8-28405-07/libmscordaccore.so
@(#)No version information produced
@(#)No version information produced
@(#)No version information produced
@(#)No version information produced
@(#)No version information produced
/usr/lib64/dotnet/shared/Microsoft.NETCore.App/3.0.0-preview8-28405-07/libmscordbi.so
@(#)No version information produced
/usr/lib64/dotnet/shared/Microsoft.NETCore.App/3.0.0-preview8-28405-07/libcoreclrtraceptprovider.so
/usr/lib64/dotnet/shared/Microsoft.NETCore.App/3.0.0-preview8-28405-07/System.Security.Cryptography.Native.OpenSsl.so
@(#)Version 4.700.19.40503 @Commit: 1bb52e6a3db7f3673a3825f3677b9f27b9af99aa
/usr/lib64/dotnet/shared/Microsoft.NETCore.App/3.0.0-preview8-28405-07/libdbgshim.so
@(#)No version information produced
```

A bunch of files have `@(#)No version information produced`. Mostly seems to be from coreclr?

Not sure what the deal is with `libcoreclrtraceptprovider.so, it has no commit information, not even `@(#)No version information produced`.

**Comment by crummel** [comment]
Hmm, are there any IL assemblies with missing info or are they all native?  It looks like native sourcelink is handled a bit differently so we might have missed something there.

**Comment by omajid** [comment]
@crummel Any tips on how to find version information for IL assemblies on Linux?

**Comment by crummel** [comment]
Usually I use exiftool, or just strings if that's not available.  Sorry, not at my machine right now or I would check on it myself.

**Comment by dagood** [comment]
For the Core-Setup binaries in the list (`libnethost.so`, `libhostfxr.so`, `libhostpolicy.so`), we still don't have version info in our main build, so it's not surprising it's missing for those files in source-build too: https://github.com/dotnet/core-setup/issues/7635.

The other ones do look like CoreCLR. (Only the CoreFX ones look good.)

**Comment by dseefeld** [comment]
This should be fixed in RC1.  @crummel to verify

**Comment by omajid** [comment]
3.0 is out of support and the problem appears to be fixed on 3.1:

```
$ /usr/lib64/dotnet/dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   3.1.104
 Commit:    0cf4df563a

Runtime Environment:
 OS Name:     fedora
 OS Version:  32
 OS Platform: Linux
 RID:         fedora.32-x64
 Base Path:   /usr/lib64/dotnet/sdk/3.1.104/

Host (useful for support):
  Version: 3.1.4
  Commit:  0c2e69caa6

.NET Core SDKs installed:
  3.1.104 [/usr/lib64/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.1.4 [/usr/lib64/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.4 [/usr/lib64/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
$ find /usr/lib64/dotnet -iname '*.so' -exec sh -c "echo {} ; strings {} | grep '@(#)'" \; | grep -i unknown
$ echo $?
1
$ find /usr/lib64/dotnet -iname '*.so' -exec sh -c "echo {} ; strings {} | grep '@(#)'" \; | grep -i No
$ echo $?
1
```
